### PR TITLE
Heart tab filters statuses for all collapses

### PR DESCRIPTION
**Pull request by Tyroie**
I tried out a couple ideas for this, and I think this one makes the most sense -
In the heart tab only, any filters for which there are already collapses are removed. `Friends`, `Bookmarks`, `Mods`, `Interested`, `Not Interested` and `Ignored`. That just leaves the Gender and Status filters.

Those other filters didn't seem very useful to me, and I couldn't find a way to make them work that wasn't buggy in another way. The shorter filter list is quicker & easier to use, I think.

The main use of course is that now the status filters function for all lists properly! Very nice for finding friends and bookmarks who are Looking.

### CODE REVIEW

**Comment by WreckedAvent** [comment]
so what's going on? Why did reducing the number of options fix ... anything?

**Comment by Tyroie** [comment]
The TLDR is that it seems to me that in the Heart tab, if you want to filter out Friends, just close the Friends collapse. Though that's oversimplifying it of course.

The initial problem is pretty simple - the `Friends`, `Bookmarks` etc lists were only affected by their own filter, `Friends filter` `Bookmarks filter` etc. The other ones did nothing for those lists. Only `Search Results` functioned completely well.

My first idea was to make each list ignore its own filter completely. This worked mostly well. If you unchecked all status filters, then you could see which friends are also bookmarks, etc. Though I only personally see it as marginally useful.

However if say `Online status` is checked and `Friends` is unchecked, then you see all Bookmarks who are Online except Friends. This behavior seems useless at best, to me, it kind of rubbed me the wrong way.

I *could* give the Heart tab its own filtering code, where statuses take priority before the other filters. It's the best alternative I can think of.

Let me know how you'd like it done, I'll do it that way!

**Comment by WreckedAvent** [comment]
Hmmm, I suppose we don't really need the friend checkmarks and others on the heart tab anyway, as they're already organized appropriately in the dropdowns.

Approved. :100:
